src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs

```diff
@@ -338,6 +354,12 @@ private DriveInfo ReportCommonDiskUsage(string dirType, string directory)
                 }
                 return true;
             }
+            catch (PathTooLongException)
```

Comment:
Are you just completely not trying to handle pathtoolongexception? Some repos, like core-setup, currently have long path issues